### DISCUSSION TITLE
ENH: Save context of objects in backend and leverage in UX

### DISCUSSION THREAD

**Discussion by GaelVaroquaux** [probabl-ai/skore#577] [Ideas]
Summarizing here a discussion on how to build a UX that helps searching and giving meaning / context to the objects stored. The idea being that when an object is inserted in the store it is useful to save context with the insertion as it is hard to understand or search a long list of stored of objects, especially over a long period of time.

![xkcd](https://imgs.xkcd.com/comics/documents.png)

Some ideas (non exhaustive) on meta data to add to objects stored:

- Line of code (file, and line number) of the call that stored the object
- Identifier of the executable in which the call was made (pid, command line, which will then be analyzed to be turned into something human understandable, eg "jupyter" vs "vscode")
- The "sys.prefix", which will be analyzed for instance to extract which virtual environment was used
- If we are in a git, the current git hash
- Hash of data (X, y) associated to a result
- Identifier of the user that made the call
- Identifier of the computer on which the call was made (and probably a bit of information on it: hardware & OS)

The UX (front) will later be able to leverage these to facilitate searching and understanding objects

**Comment by tuscland** [comment]
I would love to see this implemented fast so we can start having ideas on how to leverage context. Maybe some preparative thinking on #153 could help.

**Comment by tuscland** [comment]
We could also save the versions of Python dependencies so that it helps detecting (àla bisect) sudden changes in model performance.
